modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/roles.jsp

```diff
@@ -145,6 +145,29 @@ currentURLObj.setParameter("historyKey", liferayPortletResponse.getNamespace() +
 
 					var searchContainer = Liferay.SearchContainer.get(searchContainerName);
 
+					let searchContainerData = searchContainer.getData(true);
+
+					<%
+					long[] roleIds = new long[0];
+
+					if (selUser != null) {
+						roleIds = selUser.getRoleIds();
+					}
+
+					JSONArray roleIdsJSONArray = JSONFactoryUtil.createJSONArray();
+
+					for (int i = 0; i < roleIds.length; i++) {
+						roleIdsJSONArray.put(roleIds[i]);
+					}
```

Comment:
probably is easier to read something like this
```
Long[] roleIds = new Long[0];
							
if (selUser != null) {
		roleIds = ArrayUtil.toArray(selUser.getRoleIds());
}

JSONArray roleIdsJSONArray = JSONFactoryUtil.createJSONArray(roleIds);
```

Reply:
I don't get why 
```
for (int i = 0; i < roleIds.length; i++) {
		roleIdsJSONArray.put(roleIds[i]);
	}
```
should leave the roleIdsJSONArray as a "string array".
And what about
```
String[] roleIds = new String[0]; 

if (selUser != null) { 
  roleIds = ArrayUtil.toStringArray(selUser.getRoleIds()); 
  } 
  
  JSONArray roleIdsJSONArray = JSONFactoryUtil.createJSONArray(roleIds);
```
?